File: 1561-141-1400-construct-k-palindrome-strings/1561-141-1400-construct-k-palindrome-strings.js
var canConstruct = function(s, k) {
    if (k > s.length) {
        return false;
    }

    let freq = 0;
    for (const ch of s) {
        freq ^= (1 << (ch.charCodeAt(0) - 'a'.charCodeAt(0)));
    }

    return freq.toString(2).split('1').length - 1 <= k;
};

File: 1561-141-1400-construct-k-palindrome-strings/1561-141-1400-construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if k > len(s):
            return False
        
        freq = 0
        for ch in s:
            freq ^= (1 << (ord(ch) - ord('a')))
        
        return bin(freq).count('1') <= k
